### PR TITLE
Split resource struct to dynamic and static ones

### PR DESCRIPTION
**Pull request by anttiylitokola**

### CODE REVIEW

File: nsdl-c/sn_nsdl_lib.h [hunk 1]
```diff
@@ -34,7 +34,7 @@ extern "C" {
 #define SN_NSDL_ENDPOINT_NOT_REGISTERED  0
 #define SN_NSDL_ENDPOINT_IS_REGISTERED   1
 
-//#define MEMORY_OPTIMIZED_API
+#define MEMORY_OPTIMIZED_API
```
Comment (TeroJaasko) [inline nsdl-c/sn_nsdl_lib.h]: Is this related to this change?

**Comment by TeroJaasko** [comment]
Since the bootstrap functionality is being removed, it might be good to have some explanation of reasoning (duplicate functionality with the one in mbed-client?) of it just for the record.

**Comment by TeroJaasko** [comment]
Would it be possible to separate these commits logically, ie. have the lwm2m bootstrap removal in its own commit and the structure split in the another? The "Split resource struct to dynamic and static.." commit is also removing eg. omalw_certificate_list_t, sn_nsdl_get_certificates, sn_nsdl_update_certificates and sn_nsdl_create_oma_device_object().

**Comment by anttiylitokola** [comment]
Made another PR for splitting the commits. https://github.com/ARMmbed/mbed-client-c/pull/94
